### ISSUE TITLE
Android bundle process stuck at createBundleReleaseJsAndAssets

### ISSUE THREAD

**Issue by tofujlg** [nativewind/nativewind#1150]
**Describe the bug**
I run `npx react-native build-android --mode=release` command to generate `abb` file so that I can upload the app to the playStore.

The build process gets stuck at `:app:createBundleReleaseJsAndAssets` and cannot generate an app bundle.

![PXL_20241016_223946537](https://github.com/user-attachments/assets/29f0e86d-cba8-4d43-86b1-30ded4b72e22)

 I waited up to 8 hours but there was no progress.

The above command is the official commend from [react native doc](https://reactnative.dev/docs/signed-apk-android) but we often use `gradlew` command to generate apk or abb. 
I also tried `./gradlew clean && ./gradlew assembleRelease` but the result was the same.

It seems like the issue is coming from `withNativeWind` function in `metro.config.js` file. 

```js
const { withNativeWind } = require('nativewind/metro');
//...
module.exports = withNativeWind(config, { input: './global.css' });
```
If I modify the final line to this(which is expo default config without nativeWind), the build process succeeds but of course ,there are many styling bugs.

```js
module.exports = config;
```

**Reproduction**
Initially, this happened in my project where I installed rnr along side with nativeWind but I was able to reproduce the bug in `create-expo-stack` project.

Here is [the reproduction repo](https://github.com/tofujlg/nwtest). 
And here is the command with which I generated the repo :

```shell
  npx create-expo-stack@latest nwtest --expo-router --nativewind
```

**Expected behavior**
The build process should be completed without issue.

**Additional context**
Thanks for building amazing stuff.

**Comment by krishna-gujjjar** [comment]
I am facing same issue. downgrade nativewind to `v4.1.10` or `v4.1.7`
The latest version stuck after:

```sh
Copying 28 asset files
Done writing bundle output
Done writing sourcemap output

<=------------> 7% EXECUTING [1m 22s]
> :app:createBundleReleaseJsAndAssets
```

**Comment by tofujlg** [comment]
@krishna-gujjjar 
Thanks for the info. I downgraded to `4.1.10` and now I can build!
I will work on this ver for now.

**Comment by tofujlg** [comment]
Update: 
I was able to build apk and istalled it on my Pixel 7 but I get the following error and app does not work.

```log
10-18 09:25:10.163 27864 27907 D SoLoader: libturbomodulejsijni.so found on /data/app/~~ngIM4kf9fbmDxI5IZlgbWg==/com.anonymous.groovie-kUGaFjNLr8Lz-CGKUABfiQ==/base.apk!/lib/arm64-v8a
10-18 09:25:10.165 27864 27907 I ExpoModulesCore: ✅ JSI interop was installed
10-18 09:25:10.165 27864 27907 W ExpoModulesCore: ⚠️ JSI interop was already installed
10-18 09:25:10.166 27864 27907 I ExpoModulesCore: ✅ Constants were exported
10-18 09:25:10.183 27864 27907 D nativeloader: Load /data/app/~~ngIM4kf9fbmDxI5IZlgbWg==/com.anonymous.groovie-kUGaFjNLr8Lz-CGKUABfiQ==/base.apk!/lib/arm64-v8a/librnscreens.so using ns clns-4 from class loader (caller=/data/app/~~ngIM4kf9fbmDxI5IZlgbWg==/com.anonymous.groovie-kUGaFjNLr8Lz-CGKUABfiQ==/base.apk!classes2.dex): ok
10-18 09:25:10.184 27864 27907 I ReactNativeJS: Running "main
10-18 09:25:10.193 27864 27907 V AsyncStorageExpoMigration: No scoped database found
10-18 09:25:10.196 27864 27908 D CompatibilityChangeReporter: Compat change id reported: 289878283; UID 10430; state: ENABLED
10-18 09:25:10.240 27864 27907 E ReactNativeJS: Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined.
10-18 09:25:10.240 27864 27907 E ReactNativeJS:
10-18 09:25:10.240 27864 27907 E ReactNativeJS: This error is located at:
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Suspense
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Route
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in StaticContainer
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in EnsureSingleNavigator
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in SceneView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RCTView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RCTView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Background
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Screen
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RNSScreen
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Suspender
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Suspense
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Freeze
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in DelayedFreeze
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in InnerScreen
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Screen
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in MaybeScreen
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RNSScreenContainer
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in ScreenContainer
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in MaybeScreenContainer
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RCTView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in SafeAreaProviderCompat
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in BottomTabView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in PreventRemoveProvider
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in NavigationContent
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in BottomTabNavigator
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in TabLayout
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Suspense
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Route
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in StaticContainer
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in EnsureSingleNavigator
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in SceneView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RCTView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in DebugContainer
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in MaybeNestedStack
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RCTView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RNSScreen
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Suspender
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Suspense
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Freeze
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in DelayedFreeze
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in InnerScreen
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Screen
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in SceneView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Suspender
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Suspense
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Freeze
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in DelayedFreeze
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RNSScreenStack
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in ScreenStack
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in NativeStackViewInner
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RCTView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in SafeAreaProviderCompat
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in NativeStackView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in PreventRemoveProvider
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in NavigationContent
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in NativeStackNavigator
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in ThemeProvider
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RootLayout
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Try
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Suspense
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Route
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RNCSafeAreaProvider
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in SafeAreaProvider
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in wrapper
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in EnsureSingleNavigator
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in BaseNavigationContainer
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in ThemeProvider
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in NavigationContainerInner
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in ContextNavigator
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in ExpoRoot
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in App
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in RCTView
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in Unknown
10-18 09:25:10.240 27864 27907 E ReactNativeJS:     in AppContainer, js engine: hermes
10-18 09:25:10.245 27864 27864 D CompatibilityChangeReporter: Compat change id reported: 210923482; UID 10430; state: ENABLED
10-18 09:25:10.245 27864 27864 D CompatibilityChangeReporter: Compat change id reported: 63938206; UID 10430; state: DISABLED
10-18 09:25:10.246 27864 27908 E unknown:ReactNative: Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined.
```

**Comment by marklawlor** [comment]
Fixed in `4.1.18`. If you are still getting this issue please create a new issue with a production and detailed instructions on how you build your app.

**Comment by tofujlg** [comment]
@marklawlor 
Updated to `4.1.18` and confirmed that I can build apk. 
Thank you veruy much !
